libs/sync/src/lib/add-sync-page/add-sync-page.component.ts

```diff
@@ -0,0 +1,275 @@
+  step = computed(() => this.stepIndex() + 1);
+
+  validForm = false;
+  configuration?: ISyncEntity;
+  folderSelection: SyncItem[] = [];
+  loading = false;
+  saving = false;
+
+  get backPath() {
+    return this.syncId ? '../../..' : '../..';
+  }
+
+  ngOnInit() {
+    this.currentRoute.params
+      .pipe(
+        filter((params) => !!params['syncId']),
+        take(1),
+        tap((params) => {
+          this.loading = true;
+          this.cdr.markForCheck();
+          this.syncId = params['syncId'] as string;
+          this.syncService.setCurrentSyncId(this.syncId);
+          this.validForm = true;
+        }),
+        switchMap(() => this.syncService.getSync(this.syncId as string)),
+        tap((sync) => {
+          // We keep the distinction between loaded sync and configuration from the form to make sure the configurationForm component
+          // will trigger the sync setter only once
+          this.sync = sync;
+          this.configuration = sync;
+          this.cdr.markForCheck();
+        }),
+        switchMap((sync) => {
+          return this.syncService.hasCurrentSourceAuth().pipe(
+            tap((hasAuth) => {
+              if (hasAuth) {
+                this.loading = false;
+                this.cdr.markForCheck();
+              }
+            }),
+            filter((hasAuth) => !hasAuth),
+            map(() => this.syncService.getConnector(sync.connector.name, sync.id)),
+            switchMap((connector) => {
+              return this.syncService.authenticateToConnector(sync.connector.name, connector);
+            }),
+          );
+        }),
+      )
+      .subscribe({
+        next: () => {
+          this._goNext();
+          this.loading = false;
+          this.cdr.markForCheck();
+          this.router.navigate([], { queryParams: {} });
+        },
+        error: () => this.toaster.error('sync.details.authentication.fail'),
+      });
+  }
+
+  goBack() {
+    if (this.stepIndex() > 0) {
+      this.stepIndex.update((value) => value - 1);
+    }
+  }
+
+  goNext() {
+    if (!this.configuration) {
+      return;
+    }
+    const syncEntity = this.configuration;
+    if (!this.syncId) {
+      this.connectorDefinition
+        .pipe(
+          take(1),
+          switchMap(
+            (connectorDef) =>
+              this.modalService.openConfirm({
+                title: this.translate.instant('sync.add-page.confirm-authenticate.title', {
+                  connector: connectorDef.title,
+                }),
+                description: this.translate.instant('sync.add-page.confirm-authenticate.description'),
+                confirmLabel: this.translate.instant('sync.add-page.confirm-authenticate.confirm-button'),
+              }).onClose,
+          ),
+          filter((confirmed) => !!confirmed),
+          switchMap(() => this._createSync(syncEntity)),
+        )
+        .subscribe({
+          next: (connector) => this._onSuccessfulCreation(connector, syncEntity),
```

Comment:
subscribe in subscribe (`_onSuccessfulCreation` calls `_syncCreationDone` which makes a subscribe)

Reply:
right, I added `triggerSync` at the end and didn't think about that 🤦 
Fixing